awx/ui_next/src/components/LaunchPrompt/CredentialsStep.jsx

```diff
@@ -1,7 +1,172 @@
-import React from 'react';
+import React, { useState, useCallback, useEffect } from 'react';
+import { useHistory } from 'react-router-dom';
+import { withI18n } from '@lingui/react';
+import { t } from '@lingui/macro';
+import { useField } from 'formik';
+import { ToolbarItem } from '@patternfly/react-core';
+import { CredentialsAPI, CredentialTypesAPI } from '@api';
+import AnsibleSelect from '@components/AnsibleSelect';
+import OptionsList from '@components/OptionsList';
+import ContentLoading from '@components/ContentLoading';
+import CredentialChip from '@components/CredentialChip';
+import ContentError from '@components/ContentError';
+import { getQSConfig, parseQueryString } from '@util/qs';
+import useRequest from '@util/useRequest';
 
-function CredentialsStep() {
-  return <div />;
+const QS_CONFIG = getQSConfig('inventory', {
+  page: 1,
+  page_size: 5,
+  order_by: 'name',
+});
+
+function CredentialsStep({ i18n }) {
+  const [field, , helpers] = useField('credentials');
+  const [selectedType, setSelectedType] = useState(null);
+  const history = useHistory();
+
+  const isTypeSelected = !!selectedType;
+  const {
+    result: types,
+    error: typesError,
+    isLoading: isTypesLoading,
+    request: fetchTypes,
+  } = useRequest(
+    useCallback(async () => {
+      const loadedTypes = await CredentialTypesAPI.loadAllTypes();
+      if (!isTypeSelected && loadedTypes.length) {
+        const match =
+          loadedTypes.find(type => type.kind === 'ssh') || loadedTypes[0];
+        setSelectedType(match);
+      }
+      return loadedTypes;
+    }, [isTypeSelected]),
+    []
+  );
+
+  useEffect(() => {
+    fetchTypes();
```

Comment:
The credential types request gets called twice when I navigate to the credentials step. It looks like `isTypeSelected` initializes as false because `selectedType` is null, then `fetchTypes` gets called and sets `selectedType` to a value, and finally `selectedType` value changes which triggers `fetchTypes` again.

Reply:
good catch. Fixed -- I realized I can assume there is no type selected after fetching, since there haven't been any to select from at that point